xc/device.ml

```diff
@@ -603,7 +603,21 @@ let qemu_media_change ~xs device _type params =
 		"params",         pathtowrite;
 	] in
 	Xs.transaction xs (fun t -> t.Xst.writev backend_path back_delta);
-	debug "Media changed: params = %s" pathtowrite
+	debug "Media changed: params = %s" pathtowrite;
+
+	try
+		ignore(with_xs (fun xs -> xs.Xs.read (Printf.sprintf "/libxl/%d/dm-version" device.frontend.domid)) = "qemu_xen");
+		let c = Qmp_protocol.connect (Printf.sprintf "/var/run/xen/qmp-libxl-%d" device.frontend.domid) in
```

Comment:
I don't like how we do the discovery right now, but I think this can go in while a more global decision on the design is taken